lib/strava/web/request.rb

```diff
@@ -35,7 +35,23 @@ def request(method, path, options)
           end
           request.options.merge!(options.delete(:request)) if options.key?(:request)
         end
-        response.body
+
+        Strava::Web::ResponseWrap.new(conditional_response_upgrade(response))
+      end
+
+      def conditional_response_upgrade(response)
+        case response.body.class.name
```

Comment:
Does it work with just `.class` and `case Array`? Avoids conversion to string. I would just use `if` and `is_a?`.